File: 08_Practice_Problem_2.cpp
/*Write a C++ program to create an Employee class with data members: 
Employee number, Employee name, Basic, DA, IT, Net Salary class has 3 methods,
one method to get employee details for name, ID and basic salary
other method to calculate net salary using formula
	DA=1.32*basic;
	IT=0.30*(basic+DA);
	net_sal=(basic+DA)-IT; 
third method to print emp name, ID, net salary along with all salary components
Use the concept of array to enter data for minimum 3 Employees.*/

#include<iostream>
using namespace std;

class Employee
{
	public:
	int EmployeeNumber;
	float Basic, DA, IT, Salary;
	string EmployeeName;

	void get_details()             
	{
		cout<<"\nEnter Employee ID"<<endl;
		cin>>EmployeeNumber;
		cout<<"Enter Employee name "<<endl;
		cin>>EmployeeName;
		cout<<"Enter Employee Basic Salary"<<endl;
		cin>>Basic;
	}
	void calculate_details()       
	{
		DA=1.32*Basic;
		IT=0.30*(Basic+DA);
		Salary=(Basic+DA)-IT;	
	}
	void print_details()           
	{
		cout<<"\nEmployee Id :"<<EmployeeNumber<<endl;
		cout<<"Employee name : "<<EmployeeName<<endl;
		cout<<"Employee Basic Salary : "<<Basic<<endl;
		cout<<"Employee DA : "<<DA<<endl;
		cout<<"Employee IT : "<<IT<<endl;
		cout<<"Employee Salary : "<<Salary<<endl;
	}
};
   
    int main()
    {
    Employee E1[3];
    for(int i=0 ; i<3; i++)
    {
    	cout<<"Enter details of Employee: "<<i+1<<endl;
    	E1[i].get_details();
    	E1[i].calculate_details();
        E1[i].print_details();
        cout<<""<<endl;
	}
    return 0;
}

File: 08_Practice_Problem_2.java
/*Write a C++ program to create an Employee class with data members: 
Employee number, Employee name, Basic, DA, IT, Net Salary class has 3 methods,
one method to get employee details for name, ID and basic salary
other method to calculate net salary using formula
	DA=1.32*basic;
	IT=0.30*(basic+DA);
	net_sal=(basic+DA)-IT; 
third method to print emp name, ID, net salary along with all salary components
Use the concept of array to enter data for minimum 3 Employees.*/

package extc1;
import java.util.Scanner;

public class employee {
    Scanner in = new Scanner(System.in);
    int EmployeeNumber;    
    String EmployeeName;
    double Basic, DA, IT, Salary;
    
    public employee() {
    }

    void get_details() {
        System.out.println("Enter the Id of Employee : ");
        EmployeeNumber = in.nextInt();
        System.out.println("Enter the name of Employee : ");
        EmployeeName = in.next();
        System.out.println("Enter the Basic Salary of Employee");
        Basic = in.nextFloat();
    }

    void calculate_details() {
        DA = 1.32 * (Basic);
        IT = 0.30 * (Basic + DA);
        Salary = (Basic + DA) - IT;
    }

    void print_details() {
        System.out.println("Employ Id" + EmployeeNumber);
        System.out.println("Employ name : " + EmployeeName);
        System.out.println("Employ Basic Salary : " + Basic);
        System.out.println("Employ DA : " + DA);
        System.out.println("Employ IT : " + IT);
        System.out.println("Employ Salary : " + Salary);
    }

    public static void main(String[] args) {
    	employee[] obj = new employee[2];
        for (int i = 0; i < 2; i++) {
        	System.out.println("Enter details of Employee:"+(i+1));
        	obj[i]= new employee();
            obj[i].get_details();
            obj[i].calculate_details();
        }
        for (int i = 0; i < 2; i++) {
        	obj[i].print_details();
        }
    }
}
